### PR TITLE
chore: update `props` types in Text and Box component usages

### PR DESCRIPTION
**Pull request by davidmurdoch**
## **Description**

I'm updating the codebase to TypeScript 5.3.3 and in v5.3.3 this code would not pass the typechecker. Now it does 🌞 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by HowardBraham — DISMISSED** [review]

File: ui/components/component-library/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ export const Checkbox: CheckboxComponent = React.forwardRef(
       title,
       name,
       label,
+      as: _as, // ignore, `as` in props is not a valid property on `Box`
```
Comment (davidmurdoch) [inline ui/components/component-library/checkbox/checkbox.tsx:35]:
```suggestion
      as: _as, // ignore, `as` in props is not a valid property on `Text`
```

**Comment by brad-decker** [comment]
@georgewrmarshall or @garrettbear wanna give this a thumbs up / request changes?

**Review by HowardBraham — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]
Okay looks good to me. 

Just clarifying that `underscore prefix in TypeScript is used to indicate that a variable or property should be ignored`

File: ui/components/component-library/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -56,7 +57,7 @@ export const Checkbox: CheckboxComponent = React.forwardRef(
         alignItems={AlignItems.center}
         ref={ref}
         htmlFor={id}
-        {...props} // TODO: There is a typing issue with spreading props to the Box component. It still works but TypeScript complains.
+        {...props}
```
Comment (georgewrmarshall) [inline ui/components/component-library/checkbox/checkbox.tsx:60]:
```suggestion
        ...(props as Checkbox<C>)
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
I think we use the suggested conventions for the `as` prop. These components were missing them. Does this also work?

**Comment by davidmurdoch** [comment]
> Just clarifying that `underscore prefix in TypeScript is used to indicate that a variable or property should be ignored`

It's a general convention that indicates that the variable is intentionally unused -- not specific to TypeScript. Most tooling respects the prefix.

File: ui/components/component-library/checkbox/checkbox.tsx (hunk 2 above)
Comment (davidmurdoch) [reply ui/components/component-library/checkbox/checkbox.tsx:60]: This already has an `as="label"` property, do we want to continue to override it?

Comment (davidmurdoch) [reply ui/components/component-library/checkbox/checkbox.tsx:60]: I also don't really see how this would work.

Comment (georgewrmarshall) [reply ui/components/component-library/checkbox/checkbox.tsx:60]: Ah apologies. I see your point. I'll look into it further

File: ui/components/component-library/badge-wrapper/badge-wrapper.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import classnames from 'classnames';
 
 import { Display } from '../../../helpers/constants/design-system';
 import { Box } from '..';
-import type { PolymorphicRef } from '../box';
+import type { BoxProps, PolymorphicRef } from '..';
```
Comment (georgewrmarshall) [inline ui/components/component-library/badge-wrapper/badge-wrapper.tsx:6]: now exported from root for easier DX

File: ui/components/component-library/badge-wrapper/badge-wrapper.tsx [hunk 4]
```diff
@@ -30,7 +30,7 @@ export const BadgeWrapper: BadgeWrapperComponent = React.forwardRef(
       className={classnames('mm-badge-wrapper', className)}
       ref={ref}
       display={Display.InlineBlock}
-      {...props} // TODO: There is a typing issue with spreading props to the Box component. It still works but TypeScript complains.
+      {...(props as BoxProps<C>)}
```
Comment (georgewrmarshall) [inline ui/components/component-library/badge-wrapper/badge-wrapper.tsx:33]: passes spread props as box props but also retains polymorphic prop types

File: ui/components/component-library/badge-wrapper/badge-wrapper.types.ts [hunk 5]
```diff
@@ -29,7 +30,7 @@ export interface BadgeWrapperStyleUtilityProps extends StyleUtilityProps {
   /**
    * The BadgeWrapper props of the component. All Box props can be used
    */
-  badgeContainerProps?: any; // TODO: Replace with Box types when the syntax and typing is properly figured out. Needs to accept everything Box accepts
+  badgeContainerProps?: BoxProps<'div'>;
```
Comment (georgewrmarshall) [inline ui/components/component-library/badge-wrapper/badge-wrapper.types.ts:33]: FIxing types here but not allowing root element to change

File: ui/components/component-library/checkbox/checkbox.tsx [hunk 6]
```diff
@@ -51,12 +51,11 @@ export const Checkbox: CheckboxComponent = React.forwardRef(
         className={classnames('mm-checkbox', className, {
           'mm-checkbox--disabled': Boolean(isDisabled),
         })}
-        as="label"
         display={Display.InlineFlex}
         alignItems={AlignItems.center}
         ref={ref}
         htmlFor={id}
-        {...props} // TODO: There is a typing issue with spreading props to the Box component. It still works but TypeScript complains.
+        {...(props as TextProps<C>)}
```
Comment (georgewrmarshall) [inline ui/components/component-library/checkbox/checkbox.tsx:58]: Fixing spread props here to ensure they are Text typed

File: ui/components/component-library/index.ts [hunk 7]
```diff
@@ -22,7 +22,13 @@ export {
...
 export type { BadgeWrapperProps } from './badge-wrapper';
 export { Box } from './box';
-export type { BoxProps, StyleUtilityProps } from './box';
+export type {
+  BoxProps,
+  BoxComponent,
+  StyleUtilityProps,
+  PolymorphicRef,
+  PolymorphicComponentPropWithRef,
+} from './box';
```
Comment (georgewrmarshall) [inline ui/components/component-library/index.ts:31]: Exporting more types from root for easier importing

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thank you @davidmurdoch for improving typescript for the extension 🙏

File: ui/components/component-library/badge-wrapper/badge-wrapper.tsx [hunk 8]
```diff
@@ -30,7 +31,7 @@ export const BadgeWrapper: BadgeWrapperComponent = React.forwardRef(
       className={classnames('mm-badge-wrapper', className)}
       ref={ref}
       display={Display.InlineBlock}
-      {...props} // TODO: There is a typing issue with spreading props to the Box component. It still works but TypeScript complains.
+      {...props}
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/badge-wrapper.tsx]: I messed up on this suggestion apologies! Been out of the component game too long 😓

File: ui/components/component-library/checkbox/checkbox.tsx (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/component-library/checkbox/checkbox.tsx]: Apologies on the lazy ill-informed initial review again 🙇

**Review by garrettbear — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**